### PR TITLE
Development

### PR DESCRIPTION
**Pull request by KhalidZayat**
update readme.md file
adding isBefore and is after methods
create DateCharacteristcs struct to collect Date characteristics in
adding test function of isBefore and is After

### CODE REVIEW

File: DateTransformer/DateFormater.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+class DateTransformer {
+    
+    /// Convert date to a string using custom date format.
+    ///
+    /// - Parameters:
+    /// - timeStamp: timeStamp value to be transformed to date
+    /// - characteristcs: format of the string representation, calendar identifre,  custom locale.
+    /// - Returns: string representation of the date.
+    static func dateFromTimeStamp(timeStamp: Double, characteristcs: DateCharacteristcs) -> String {
```
Comment (IbrahimAli2017) [inline DateTransformer/DateFormater.swift]: rename function to be formatTimeStamp

File: DateTransformer/DateFormater.swift [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    /// Convert string with custom format To string using custom date format.
+    ///
+    /// - Parameters:
+    /// - string: string to be transformed
+    /// - format: format of the string representation.
+    /// - locale: to fix a custom locale.
+    /// - identifre: calendar identifre (.gregorian, .islamic, etc...).
+    /// - Returns: string representation of the date.
+    static func stringToString(string: String, inputCharacteristcs: DateCharacteristcs, outputCharacteristcs: DateCharacteristcs)-> String {
+        var formatter = DateFormatter(characteristcs: inputCharacteristcs)
```
Comment (IbrahimAli2017) [inline DateTransformer/DateFormater.swift]: should call the previous functions
stringToDate
DateToString

File: DateTransformerTests/DateTransformerTests.swift [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        return Calendar(identifier: .gregorian).date(from: dateComponents)!
+    }
+    
+    //only year
+    func testYearFormat_passed() {
+        XCTAssertEqual(DateTransformer.dateToString(date: mockDate, characteristcs: DateCharacteristcs(format: DateFormats.YYYY.rawValue)), "2022")
+    }
+    
+    func testYearFormat_failed() {
+
```
Comment (IbrahimAli2017) [inline DateTransformerTests/DateTransformerTests.swift]: remove empty line

File: DateTransformer/DateFormater.swift (hunk 1 above)
Comment (KhalidZayat) [reply DateTransformer/DateFormater.swift]: done and a new function to get date from timeStamp is implemented.

File: DateTransformer/DateFormater.swift (hunk 2 above)
Comment (KhalidZayat) [reply DateTransformer/DateFormater.swift]: done

File: DateTransformerTests/DateTransformerTests.swift (hunk 3 above)
Comment (KhalidZayat) [reply DateTransformerTests/DateTransformerTests.swift]: done

**Review by IbrahimAli2017 — APPROVED** [review]

**Review by IbrahimAli2017 — APPROVED** [review]

**Review by IbrahimAli2017 — APPROVED** [review]

**Review by IbrahimAli2017 — APPROVED** [review]

**Pull request merged**